siyavula/latex2image/utils.py

```diff
@@ -46,3 +50,91 @@ def copy_if_newer(src, dest):
         success = False
 
     return success
+
+
+def unescape(text):
+    '''Formats unicode characters correctly'''
+    def fixup(m):
+        text = m.group(0)
+        if text[:2] == "&#":
+            # character reference
+            try:
+                if text[:3] == "&#x":
+                    return unichr(int(text[3:-1], 16))
+                else:
+                    return unichr(int(text[2:-1]))
+            except ValueError:
+                pass
+        else:
+            # named entity
+            try:
+                text = unichr(htmlentitydefs.name2codepoint[text[1:-1]])
+            except KeyError:
+                pass
+        return text  # leave as is
+    return re.sub("&#?\w+;", fixup, text)
+
+
+def cleanup_code(code):
+    ''' Removes nested math delimiters of the form \( \) inside $ $ pairs'''
+    result = re.findall(r'\$(.*?)\$', code)
+    for snippet in result:
+        newsnippet = snippet.replace(r'\(', ' ').replace(r'\)', ' ')
+        code = code.replace(snippet, newsnippet)
+
+    code = code.strip()
+
+    # if align* or {align} in code we don't need delimiters
+    if (r'align*' in code) or ('{align}' in code):
+        if code.startswith(r'\('):
+            code = code[2:]
+        if code.endswith('\)'):
+            code = code[:-2]
+
+    # remove blank lines and lines that start with %
+    newcode = []
```

Comment:
This can be replaced with two fairly simple list comprehensions:
1) One with no blank lines (and you can strip the lines it does yield at the same time)
2) One with all elements NOT starting with '%'

If you name your two results (from the list comprehensions), you won't even need the comment and lines 94 - 102 can just be these two lines.